nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/groovy/org/apache/nifi/registry/web/security/authentication/oidc/StandardOidcIdentityProviderGroovyIT.groovy

```diff
@@ -58,17 +58,29 @@ import org.junit.runners.JUnit4
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
+import java.nio.channels.SocketChannel
+
 @RunWith(JUnit4.class)
-class StandardOidcIdentityProviderGroovyTest extends GroovyTestCase {
-    private static final Logger logger = LoggerFactory.getLogger(StandardOidcIdentityProviderGroovyTest.class)
+class StandardOidcIdentityProviderGroovyIT extends GroovyTestCase {
+    private static final Logger logger = LoggerFactory.getLogger(StandardOidcIdentityProviderGroovyIT.class)
 
     private static final Key SIGNING_KEY = new Key(id: 1, identity: "signingKey", key: "mock-signing-key-value")
 
+    private static int getAvailablePort() throws IOException {
+        SocketChannel socket = SocketChannel.open()
+        socket.setOption(StandardSocketOptions.SO_REUSEADDR, true)
+        socket.bind(new InetSocketAddress("localhost", 0))
+        return socket.socket().getLocalPort()
+    }
```

Comment:
The `NetworkUtils.getAvailableTcpPort()` in `nifi-commons` performs this function and also closes the socket after returning.  Recommend leveraging that class for this unit test.

Reply:
Unfortunately, Groovy doesn't like the try-with-resources call in that method.  However, this helped me track down the Connection timeout issue!  I just needed to close the socket.